crates/orderbook/src/api/order_validation.rs

```diff
@@ -815,4 +829,72 @@ mod tests {
             order.order_creation.fee_amount
         );
     }
+
+    #[tokio::test]
+    async fn allows_insufficient_allowance_and_balance_for_presign_orders() {
+        macro_rules! assert_allows_failed_transfer {
+            ($err:ident) => {
+                let mut fee_calculator = MockMinFeeCalculating::new();
+                let mut bad_token_detector = MockBadTokenDetecting::new();
+                let mut balance_fetcher = MockBalanceFetching::new();
+                fee_calculator
+                    .expect_get_unsubsidized_min_fee()
+                    .returning(|_, _, _| Ok(Default::default()));
+                bad_token_detector
+                    .expect_detect()
+                    .returning(|_| Ok(TokenQuality::Good));
+                balance_fetcher
+                    .expect_can_transfer()
+                    .returning(|_, _, _, _| Err(TransferSimulationError::$err));
+                let validator = OrderValidator::new(
+                    Box::new(MockCodeFetching::new()),
+                    dummy_contract!(WETH9, [0xef; 20]),
+                    vec![],
+                    Duration::from_secs(1),
+                    Arc::new(fee_calculator),
+                    Arc::new(bad_token_detector),
+                    Arc::new(balance_fetcher),
+                );
+
+                let order = OrderCreation {
+                    valid_to: u32::MAX,
+                    sell_token: H160::from_low_u64_be(1),
+                    sell_amount: 1.into(),
+                    buy_token: H160::from_low_u64_be(2),
+                    buy_amount: 1.into(),
+                    ..Default::default()
+                };
+
+                assert!(matches!(
+                    validator
+                        .validate_and_construct_order(
+                            order.clone(),
+                            None,
+                            &Default::default(),
+                            Default::default()
+                        )
+                        .await,
+                    Err(ValidationError::$err)
+                ));
```

Comment:
nit: Because SigningScheme is an enum with very few variants, it could be nice to assert an error for every SigningScheme except PreSign.